sdk/azcore/internal/exported/exported.go

```diff
@@ -110,3 +112,80 @@ func DecodeByteArray(s string, v *[]byte, format Base64Encoding) error {
 		return fmt.Errorf("unrecognized byte array format: %d", format)
 	}
 }
+
+// KeyCredential contains an authentication key used to authenticate to an Azure service.
+// Exported as azcore.KeyCredential.
+type KeyCredential struct {
+	cred *keyCredential
+}
+
+// NewKeyCredential creates a new instance of [KeyCredential] with the specified values.
+//   - key is the authentication key
+func NewKeyCredential(key string) (*KeyCredential, error) {
+	cred, err := newKeyCredential(key)
+	if err != nil {
+		return nil, err
+	}
+	return &KeyCredential{cred: cred}, nil
+}
+
+// Update replaces the existing key with the specified value.
+func (k *KeyCredential) Update(key string) error {
+	return k.cred.Update(key)
+}
+
+// SASCredential contains a shared access signature used to authenticate to an Azure service.
+// Exported as azcore.SASCredential.
+type SASCredential struct {
+	cred *keyCredential
+}
+
+// NewSASCredential creates a new instance of [SASCredential] with the specified values.
+//   - sas is the shared access signature
+func NewSASCredential(sas string) (*SASCredential, error) {
+	cred, err := newKeyCredential(sas)
+	if err != nil {
+		return nil, err
+	}
+	return &SASCredential{cred: cred}, nil
+}
+
+// Update replaces the existing shared access signature with the specified value.
+func (k *SASCredential) Update(sas string) error {
+	return k.cred.Update(sas)
+}
+
+// KeyCredentialGet returns the key for cred.
+func KeyCredentialGet(cred *KeyCredential) string {
```

Comment:
I suppose the getters aren't exported, and don't have method receivers, to prevent applications from calling them?

Reply:
Correct.  Although, to be frank, I waffled on this a bit trying to decide if we're just overengineering it.  We did apply the same pattern elsewhere (e.g. `SharedKeyCredential` in `azblob`) so there is some prior art.  And I do think there's value in not exporting APIs that aren't interesting to customers.  I will point out though that this pattern prevents SDKs from creating their own pipeline policies for these credential types.  If we think supporting that is import, then this needs to be refactored (we can always do it later in a non-breaking way).

Reply:
It seems fine - they literally pass and set the key they're updating or initializing with. Get() doesn't really provide them much value. 

Of course if someone complains we can always implement it later.